src/ripple/beast/define_print.cpp

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <beast/unit_test/amount.hpp>
+#include <beast/unit_test/global_suites.hpp>
+#include <beast/unit_test/suite.hpp>
+#include <string>
+
+// Include this .cpp in your project to gain access to the printing suite
+
+namespace beast {
+namespace unit_test {
+
+/** A suite that prints the list of globally defined suites. */
+class print_test : public suite
+{
+private:
+    template <class = void>
```

Comment:
You've probably already explained this elsewhere, but I don't usually expect the gratuitous-template-to-avoid-inlining trick in a .cpp file.  It certainly won't hurt anything, but I'm curious about the motivation.  Or am I misunderstanding the use?  Thanks.